api/app/controllers/spree/api/address_books_controller.rb

```diff
@@ -2,32 +2,41 @@ module Spree
   module Api
     class AddressBooksController < Spree::Api::BaseController
       # Note: the AddressBook is the resource to think about here, not individual addresses
+      before_filter :authorize_user_address
+      before_filter :load_user_addresses
 
       def show
-        render_address_book
+        render :show, status: :ok
       end
 
       def update
         address_params = address_book_params
         default_flag = address_params.delete(:default)
-        address = current_api_user.save_in_address_book(address_params, default_flag)
+        address = address_book_user.save_in_address_book(address_params, default_flag)
         if address.valid?
-          render_address_book
+          render :show, status: :ok
         else
           invalid_resource!(address)
         end
       end
 
       def destroy
-        current_api_user.remove_from_address_book(params[:address_id])
-        render_address_book
+        address_book_user.remove_from_address_book(params[:address_id])
```

Comment:
This isn't part of this PR but I just noticed that we don't pay attention to the return value here and maybe we should? That method returns true/false instead of raising when it fails.